File: objectiveai-sdk-py/objectiveai_sdk/functions/inventions/prompts/http.py
"""HTTP functions for invention prompt endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from objectiveai_sdk.client import ObjectiveAI
    from objectiveai_sdk.functions.inventions.prompts import (
        GetPromptResponse,
        ListPromptResponse,
        ListPromptsRequest,
        UsagePromptResponse,
    )
    from objectiveai_sdk.remote_path_commit_optional import RemotePathCommitOptional


async def list_prompts(
    client: ObjectiveAI, params: ListPromptsRequest,
) -> ListPromptResponse:
    """List all prompts accessible to the authenticated user."""
    return await client.post_unary("functions/inventions/prompts/list", params)


async def get_prompt(
    client: ObjectiveAI, params: RemotePathCommitOptional,
) -> GetPromptResponse:
    """Retrieve a prompt definition."""
    return await client.post_unary("functions/inventions/prompts", params)


async def get_prompt_usage(
    client: ObjectiveAI, params: RemotePathCommitOptional,
) -> UsagePromptResponse:
    """Retrieve usage statistics for a specific prompt."""
    return await client.post_unary("functions/inventions/prompts/usage", params)

File: objectiveai-sdk-rs/src/functions/inventions/prompts/http.rs
//! HTTP functions for invention prompt management.

use crate::{HttpClient, HttpError};

/// Lists all prompts accessible to the authenticated user.
pub async fn list_prompts(
    client: &HttpClient,
    params: super::request::ListPromptsRequest,
) -> Result<super::response::ListPromptResponse, HttpError> {
    client
        .send_unary(
            reqwest::Method::POST,
            "functions/inventions/prompts/list",
            Some(params),
        )
        .await
}

/// Retrieves a prompt definition.
pub async fn get_prompt(
    client: &HttpClient,
    params: super::request::GetPromptRequest,
) -> Result<super::response::GetPromptResponse, HttpError> {
    client
        .send_unary(
            reqwest::Method::POST,
            "functions/inventions/prompts",
            Some(params),
        )
        .await
}

/// Gets usage statistics for a specific prompt.
pub async fn get_prompt_usage(
    client: &HttpClient,
    params: super::request::GetPromptRequest,
) -> Result<super::response::UsagePromptResponse, HttpError> {
    client
        .send_unary(
            reqwest::Method::POST,
            "functions/inventions/prompts/usage",
            Some(params),
        )
        .await
}
